tests/infra/base_model_tester.py

```diff
@@ -0,0 +1,117 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Callable
+
+from flax import linen, nnx
+from transformers.modeling_flax_utils import FlaxPreTrainedModel
+
+from .base_tester import BaseTester
+from .comparison import ComparisonConfig
+from .device_runner import DeviceRunner
+from .utils import Model, Workload
+
+
+class TestType(Enum):
+    INFERENCE = "inference"
+    TRAINING = "training"
```

Comment:
We should be testing both inference and training, so we shouldn't have this enum. Training tests should just `pass` for now with a TODO in this base class to implement them later.